_posts/en/newsletters/2024-12-13-newsletter.md

```diff
@@ -0,0 +1,497 @@
+  additional risks, but a complete solution awaits additional work on
+  [package relay][topic package relay], [channel upgrades][topic channel
+  commitment upgrades], and related projects.
+
+  The vulnerability takes advantage of the LN protocol allowing an old
+  channel state to commit to a greater amount of onchain fees than the
+  fee-paying party controls in the latest state.  For example, in a
+  state where Mallory controls 99% of channel balance, she dedicates 98%
+  of overall balance to [endogenous][topic fee sourcing] fees.  She then
+  creates a new state that pays only minimal fees and forwards 99% of
+  channel balance to Bob's side.  By personally mining the old state
+  that pays 98% to fees, she can capture those fees for
+  herself---reducing the maximum value Bob can receive onchain from his
+  expected 99% to an actual 2%.  Mallory can use this method to
+  simultaneously steal from about 3,000 channels per block (with each
+  channel potentially controlled by a different victim), allowing theft
+  of about $3 million USD per block if the average channel value is
+  about $1,000 USD.
+
+  Users who became aware of the attack before losing their funds would
+  not have been able to protect themselves by shutting down their LN
+  nodes, as Mallory may have already created the necessary states.  Even
+  if the victims attempted to force close their channels in their latest
+  state (e.g., where Bob controls 99% of channel value), Mallory could
+  cause them to lose 98% of channel value by sacrificing 1% of channel
+  value herself.
+
+  The worst-case vulnerability was fixed, and the less severe variants
+  partly mitigated, by limiting the maximum amount of channel value that
+  can be dedicated to fees.  As fees in a earlier state are still
+  allowed to be higher than the amount the fee-paying party controls in
+  a later state, some theft is still possible---but the amount is
+  bounded.  A complete solution awaits improvements in fully
+  [exogenous][topic fee sourcing] fee sourcing, e.g. all states paying
+  the same commitment transaction fee, which depends on robust package
+  relay for [CPFP][topic cpfp] fee bumping in the Bitcoin P2P protocol
+  and channel commitment transaction upgrades for LN.
+
+- **Deanonymization vulnerability affecting Wasabi and related software:**
+  a developer of GingerWallet [disclosed][drkgry deanon] a method a
+  [coinjoin][topic coinjoin] coordinator could use to prevent users from
+  gaining any privacy during a coinjoin.  Bitcoin Magazine journalist
+  Shinobi [reports][shinobi deanon] that the vulnerability was
+  originally discovered in 2021 by Yuval Kogman and [reported][wasabi
+  #5439] to the Wasabi development team along with several other
+  problems.  Optech has been aware since mid-2022 that Kogman had
+  serious outstanding concerns with deployed versions of Wasabi but we
+  neglected to investigate further; we apologize to him and Wasabi
+  users for our failure.
```

Comment:
I appreciate the sincerity 🙏

Reply:
@1440000bytes For clarity, your issue in #795 appeared to be about Wasabi using Chainanalysis and it was closed on that basis.  I don't even recall it mentioning the concerns by Kogman (although it clearly does).

I became aware of Kogman's concerns when I had a call with him in July 2022 about some privacy research he was doing.  I focused on supporting his research, in the small way that I could, and didn't go back to investigate his concerns about Wasabi.  I respected him a researcher, so I should have investigated, but I was lazy and not keen to get involved in drama.